pkg/operator/ceph/csi/operator_driver.go

```diff
@@ -0,0 +1,408 @@
+					Limits:   r.Resource.Limits,
+					Requests: r.Resource.Requests,
+				}
+			case strings.Contains(r.Name, "attacher"):
+				controllerPluginResources.Attacher = &corev1.ResourceRequirements{
+					Limits:   r.Resource.Limits,
+					Requests: r.Resource.Requests,
+				}
+			case strings.Contains(r.Name, "plugin"):
+				controllerPluginResources.Plugin = &corev1.ResourceRequirements{
+					Limits:   r.Resource.Limits,
+					Requests: r.Resource.Requests,
+				}
+			case strings.Contains(r.Name, "omap-generator"):
+				controllerPluginResources.OMapGenerator = &corev1.ResourceRequirements{
+					Limits:   r.Resource.Limits,
+					Requests: r.Resource.Requests,
+				}
+			case strings.Contains(r.Name, "liveness"):
+				controllerPluginResources.Liveness = &corev1.ResourceRequirements{
+					Limits:   r.Resource.Limits,
+					Requests: r.Resource.Requests,
+				}
+			case strings.Contains(r.Name, "addons"):
+				controllerPluginResources.Addons = &corev1.ResourceRequirements{
+					Limits:   r.Resource.Limits,
+					Requests: r.Resource.Requests,
+				}
+			}
+		}
+	}
+	return controllerPluginResources
+}
+
+func createDriverNodePluginResouces(opConfig map[string]string, key string) csiopv1a1.NodePluginResourcesSpec {
+	nodePluginResources := csiopv1a1.NodePluginResourcesSpec{}
+	resource := getComputeResource(opConfig, key)
+
+	for _, r := range resource {
+		if !reflect.DeepEqual(r.Resource, corev1.ResourceRequirements{}) {
+			if strings.Contains(r.Name, "registrar") {
+				nodePluginResources.Registrar = &corev1.ResourceRequirements{
+					Limits:   r.Resource.Limits,
+					Requests: r.Resource.Requests,
+				}
+			} else if strings.Contains(r.Name, "plugin") {
+				nodePluginResources.Plugin = &corev1.ResourceRequirements{
+					Limits:   r.Resource.Limits,
+					Requests: r.Resource.Requests,
+				}
+			} else if strings.Contains(r.Name, "liveness") {
+				nodePluginResources.Liveness = &corev1.ResourceRequirements{
+					Limits:   r.Resource.Limits,
+					Requests: r.Resource.Requests,
+				}
+			} else if strings.Contains(r.Name, "addons") {
+				nodePluginResources.Addons = &corev1.ResourceRequirements{
+					Limits:   r.Resource.Limits,
+					Requests: r.Resource.Requests,
+				}
+			}
+		}
+	}
+	return nodePluginResources
+}
+
+// transferCSIDriverOwner update CSIDriver and returns the error if any
+func (r *ReconcileCSI) transferCSIDriverOwner(ctx context.Context, clusterInfo *cephclient.ClusterInfo, name string) error {
+
+	logger.Info("adding annotation to CSIDriver resource for csi-operator to own it")
+	csiDriver, err := r.context.Clientset.StorageV1().CSIDrivers().Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.Debugf("%s CSIDriver not found; skipping ownership transfer.", name)
+			return nil
+		}
+	}
+
+	key := "csi.ceph.io/ownerref"
+	ownerObjKey := client.ObjectKeyFromObject(csiDriver)
+	val, err := json.Marshal(ownerObjKey)
```

Comment:
This is the same implementation as csi operator uses to encode the ownerref? What does the json string look like?

Reply:
And the csi operator will take ownership? I guess the `Name` will be the driver name, rather than `rook-ceph`.

Reply:
yes, it is of type `types.Namespaced{Name: "*", Namespace: "*"}`